fiftyone/utils/activitynet.py

```diff
@@ -0,0 +1,1038 @@
+        labels_path (None): an optional parameter that enables explicit control
+            over the location of the labels. Can be any of the following:
+
+            -   a filename like ``"labels.json"`` specifying the location of
+                the labels in ``dataset_dir``
+            -   an absolute filepath to the labels. In this case,
+                ``dataset_dir`` has no effect on the location of the labels
+
+            If None, the parameter will default to ``labels.json``
+        compute_metadata (False): whether to produce
+            :class:`fiftyone.core.metadata.VideoMetadata` instances for each
+            video when importing
+        shuffle (False): whether to randomly shuffle the order in which the
+            samples are imported
+        seed (None): a random seed to use when shuffling
+        max_samples (None): a maximum number of samples to import. By default,
+            all samples are imported
+    """
+
+    def __init__(
+        self,
+        dataset_dir=None,
+        data_path=None,
+        labels_path=None,
+        compute_metadata=False,
+        shuffle=False,
+        seed=None,
+        max_samples=None,
+    ):
+        super().__init__(
+            dataset_dir=dataset_dir,
+            data_path=data_path,
+            labels_path=labels_path,
+            compute_metadata=compute_metadata,
+            shuffle=shuffle,
+            seed=seed,
+            max_samples=max_samples,
+        )
+
+        self.data_path = self._parse_data_path(
+            dataset_dir=dataset_dir, data_path=data_path, default="data.json",
+        )
+        self._taxonomy = None
+
+    @property
+    def has_dataset_info(self):
+        return self._classes is not None or self._taxonomy is not None
+
+    def setup(self):
+        self._sample_parser = foud.FiftyOneTemporalDetectionSampleParser(
+            compute_metadata=self.compute_metadata
+        )
+
+        self._video_paths_map = self._load_data_map(
+            self.data_path, ignore_exts=True, recursive=True
+        )
+
+        if self.labels_path is not None and os.path.isfile(self.labels_path):
+            labels = etas.load_json(self.labels_path)
+        else:
+            labels = {}
+
+        self._taxonomy = labels.get("taxonomy", None)
+        self._classes = labels.get("classes", None)
+        self._sample_parser.classes = self._classes
+        self._labels_map = labels.get("labels", {})
+        self._has_labels = any(self._labels_map.values())
+
+        uuids = sorted(self._labels_map.keys())
+        self._uuids = self._preprocess_list(uuids)
+        self._num_samples = len(self._uuids)
+
+    def get_dataset_info(self):
+        info = {}
+        if self._classes is not None:
+            info["classes"] = self._classes
+
+        if self._taxonomy is not None:
+            info["taxonomy"] = self._taxonomy
+
+        return info
+
+
+class ActivityNetDownloadConfig(object):
```

Comment:
Following up on some offline conversation- this data sharing between the 100 and 200 dataset versions has been working well for me, so I have no problem with keeping the implementation as-is